microservices/uyuni-java-common/src/main/java/com/suse/common/configuration/FileConfigurationSource.java

```diff
@@ -0,0 +1,213 @@
+     * @return the value
+     */
+    @Override
+    protected String getRawValue(String property) {
+        LOGGER.debug("getString() - called with: {}", property);
+        if (property == null) {
+            return null;
+        }
+
+        String namespace = "";
+        String propertyName = property;
+
+        int lastDot = property.lastIndexOf('.');
+        if (lastDot > 0) {
+            propertyName = property.substring(lastDot + 1);
+            namespace = property.substring(0, lastDot);
+        }
+
+        LOGGER.debug("getString() - Getting property: {}", propertyName);
+        String result = configValues.getProperty(propertyName);
+        LOGGER.debug("getString() result: {}", result);
+
+        if (result == null) {
+            if (!namespace.isEmpty()) {
+                result = configValues.getProperty(namespace + "." + propertyName);
+            }
+            else {
+                // The property was not found, and it has no namespace. Lookup all the provided fallback namespaces
+                for (String fallback : fallbackNamespaces) {
+                    result = configValues.getProperty(fallback + "." + propertyName);
+                    if (result != null) {
+                        break;
+                    }
+                }
+            }
+        }
+
+        LOGGER.debug("getString() - returning: {}", result);
+        if (result == null || result.isEmpty()) {
+            return null;
+        }
+
+        return result.trim();
+    }
+
+    @Override
+    public Set<String> getPropertyNames() {
+        return configValues.keySet().stream()
+            .map(Objects::toString)
+            .collect(Collectors.toUnmodifiableSet());
+    }
+
+    private static Stream<Path> getConfigurationFiles(Path path) {
+        if (Files.isDirectory(path)) {
+            try {
+                return Files.list(path)
+                    .filter(file -> file.getFileName().toString().endsWith(".conf"));
+            }
+            catch (IOException ex) {
+                LOGGER.error("Unable to list file in directory {}", path);
+                return Stream.empty();
+            }
+        }
+
+        return Stream.of(path);
+    }
+
+    private Properties loadProperties(Path file) {
+        Properties properties;
+
+        try {
+            String fileContent = Files.readString(file, StandardCharsets.UTF_8);
+
+            properties = new Properties();
+            properties.load(new StringReader(fileContent.replace("\\", "\\\\")));
+        }
+        catch (Exception ex) {
+            LOGGER.error("Could not parse file {}", file, ex);
+            return new Properties();
+        }
+
+        String namespace = makeNamespace(file);
+        LOGGER.debug("Adding namespace: {} for file: {}", namespace, file.toAbsolutePath());
+        for (Object key : properties.keySet()) {
+            String propertyName = (String) key;
+            if (!propertyName.startsWith(namespace)) {
```

Comment:
Not fully sure to understand. This code is meant to ensure all the properties defined in a file named for example `rhn_web.conf` starts with `web.`. In case the property does not start in the correct way, the if branch removes it from the set and then reinsert the value with the correct key. Otherwise, if the property has already the correct prefix, it is kept as is.